adsrefpipe/app.py

```diff
@@ -0,0 +1,559 @@
+
+class ADSReferencePipelineCelery(ADSCelery):
+
+    RE_PARSE_ID = re.compile(r'^H(?P<history_id>\d+)+I(?P<item_num>\d+)$')
+
+    def get_parser_name(self, source_filename):
+        """
+
+        :param source_filename:
+        :return:
+        """
+        parts = os.path.basename(source_filename).split('.')
+        source_pattern = '.%s.%s'%(parts[-2],parts[-1])
+        if source_pattern:
+            with self.session_scope() as session:
+                rows = session.query(Parser).filter(literal(source_pattern).op('~')(Parser.source_pattern)).all()
+                if len(rows) == 1:
+                    return rows[0].get_name()
+                else:
+                    self.logger.error("No unique record found in table `Parser` matching extension %s."%source_pattern)
+        else:
+            self.logger.error("Unrecognizable source file %s."%source_filename)
+        return None
+
+    def get_reference_service_endpoint(self, parsername):
+        """
+        given parsername find the endpoint that shall be called for the reference to get resolved
+
+        :param parsername:
+        :return:
+        """
+        with self.session_scope() as session:
+            rows = session.query(Parser).filter(Parser.name == parsername).all()
+            if len(rows) == 1:
+                return rows[0].get_endpoint()
+            else:
+                self.logger.error("No unique record found in table `Parser` matching name %s." % parsername)
+        return ''
+
+    def query_reference_source_tbl(self, bibcode_list=None, source_filename_list=None, parsername=None):
+        """
+        Queries reference table and returns results.
+
+        :param bibcode_list
+        :param source_filename_list
+        :param parsername
+        :return: list of json records or None
+        """
+        with self.session_scope() as session:
+            if bibcode_list and source_filename_list:
+                rows = session.query(ReferenceSource) \
+                    .filter(and_(ReferenceSource.bibcode.in_(bibcode_list),
+                                 ReferenceSource.source_filename.in_(source_filename_list))).all()
+                self.logger.info("Fetched records for bibcode = %s and source_filename = %s." % (
+                bibcode_list, source_filename_list))
+            elif bibcode_list:
+                rows = session.query(ReferenceSource).filter(ReferenceSource.bibcode.in_(bibcode_list)).all()
+                self.logger.info("Fetched records for bibcode = %s." % (bibcode_list))
+            elif source_filename_list:
+                rows = session.query(ReferenceSource).filter(ReferenceSource.source_filename.in_(source_filename_list)).all()
+                self.logger.info("Fetched records for source_filename = %s." % (source_filename_list))
+            elif parsername:
+                rows = session.query(ReferenceSource).filter(and_(ReferenceSource.parser_name == parsername)).all()
```

Comment:
Not sure where you are getting max_results from. 
So this method query_reference_source_tbl is called only from https://github.com/golnazads/ADSReferencePipeline/blob/master/run.py#L25 and it cuts any bibcode or source_filename to maximum of 5. This is only for diagnostic purposes and I doubt it would be used all that much. But the maximum number of bibcodes or source_filenames are 5, not 1000.